Bottom: 7b4509e39407700f2b83499e2db1b89ccde97650
Top:    7ab13a709c32c18a6a63e3cfe6aa048dca0c2c2a
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-08-06 15:20:06 -0700

Refresh of dup2

---

diff --git a/src/preload.c b/src/preload.c
index ff7d2d0..b3cf6ac 100644
--- a/src/preload.c
+++ b/src/preload.c
@@ -815,6 +815,7 @@ int shutdown(int socket, int how)
 int close(int socket)
 {
 	struct fd_info *fdi;
+	int ret;
 
 	init_preload();
 	fdi = idm_lookup(&idm, socket);
@@ -933,7 +934,7 @@ int dup2(int oldfd, int newfd)
 	if (!oldfdi || ret != newfd)
 		return ret;
 
-	newfdi = calloc(1, sizeof *fdi);
+	newfdi = calloc(1, sizeof *newfdi);
 	if (!newfdi) {
 		close(newfd);
 		return ERR(ENOMEM);
